### ISSUE TITLE
Imports are absolute should be relative.

### ISSUE THREAD

**Issue by mientjan** [hjeti/vue-skeleton#34]
All imports are written in a absolute way. Should be relative.

Why? 
 - Conflict with node_modules
 - Not sure where file is located
 - Will have problems in the future.
 - Conflict with eslint

Example
src/config/localeConfig.js

```js
import { PropertyNames, VariableNames } from 'data/enum/configNames';
import { getValue } from 'util/injector';
import { CONFIG_MANAGER } from 'data/Injectables';

const getLocaleConfig = () => {
  const configManager = getValue(CONFIG_MANAGER);

  const languages = configManager.getProperty(PropertyNames.AVAILABLE_LOCALES).map(locale => {
    if (typeof locale === 'string') {
      return {
        code: locale,
        urlPrefix: locale,
        translationKey: locale,
      };
    }
    return locale;
  });

  const config = {
    persistent: false,
    defaultCode: configManager.getProperty(PropertyNames.DEFAULT_LOCALE),
    languages,
  };

  return {
    localeEnabled: configManager.getVariable(VariableNames.LOCALE_ENABLED),
    localeRoutingEnabled: configManager.getVariable(VariableNames.LOCALE_ROUTING_ENABLED),
    config,
  };
};

export default getLocaleConfig;
```

**Comment by hjeti** [comment]
I don't agree on this it's now all absolute to the src folder that makes it very clear where the file is. If you make it relative you have to count the ../ to see where it is exactly.

**Comment by gianko** [comment]
Absolute is better if you have to move the component.

But with relative you can cmd+click on the IDE to go to the component.
 Also Eslint import rule `import/no-absolute-path`.

**Comment by ReneDrie** [comment]
Gianko, if you set the “src” folder to be the resource root in jetbrains, you are also able to cmd + click to the importen file.

**Comment by mientjan** [comment]
It conflicts with the eslint rule. import/no-absolute-path
this will have problems in the future we did the same thing with gaia and this had a lot of problems later on.

"Gianko, if you set the “src” folder to be the resource root in jetbrains, you are also able to cmd + click to the importen file." Not everyone is using webstorm @ReneDrie

**Comment by ThaNarie** [comment]
To resolve eslint import errors you can link to the webpack config, so that it follows the complete webpack resolve section (paths, extensions, etc).

```js
  plugins: ['import', 'prettier'],
  settings: {
    // webpack integration
    'import/resolver': {
      webpack: {
        config: 'build-tools/config/webpack/webpack.config.js',
      },
    },
  },
```

Renaming/moving is a two-edged sword.
* When renaming a folder, all relative imports within that folder are still working properly (win for relative)
* When moving a single file, all absolute imports will keep working (win for absolute)

If performance is your thing, you would favor relative imports, since it has less lookup. Absolute imports have to be looked up in both webpack source folder(s), and after that in node modules. This means that all node module imports (e.g. lodash) will first be resolved in the 'src' folder. As MJ mentioned, this could potentially result in conflicts (having a folder in your source that is similar to a node module).

I'm divided myself. For mostly technical reasons I'm a proponent of relative imports. From a UX point of view I see how they make some things clearer to reason about. A mix between relative (for max 2x ../) and absolute (where they cross their 'module boundary') would probably be 'best', but that might be even more confusing. (btw, you can configure webstorm to insert either relative or absolute imports when auto-importing classes/functions)

I've worked in quite a big project, where we had multiple 'sites' in a single repo with a 'common' folder. There we have the rule that only imports to 'common' can be absolute, all others must be relative. This makes it very easy to spot accidental cross-project imports, and it's a clear difference between project and common imports.

Absolute imports are a non-natural feature that needs a lot of configuration in bundlers, linters, compilers (TS) to make them work properly, where relative imports just work.

Just my 2 cts.

**Comment by flut1** [comment]
I would like to elaborate on some of the points that are already made by others above

@mientjan **Conflict with node_modules**
When having the `src` folder as a root, it is quite easy to accidentally create ambiguity. For example, in my current project I have a `src/config` folder. This would conflict with the `config` node module.

@mientjan **Not sure where file is located** 
Related to the above, one thing that I like about relative imports is that when you look at them it is immediately clear if they come from `node_modules` or not

@ThaNarie **I've worked in quite a big project, where we had multiple 'sites' in a single repo with a 'common' folder**
This project was set up in a microservice-like structure with a more-or-less fixed number of services. There are imports between services, but each service has their own build and deployment pipeline. In this case, it wouldn't make much sense to use relative imports for the cross-service dependencies, because that would stop us from being able to move the services.

In summary, in my opinion we should allow absolute imports, but only if there are a limited and fixed number of directories that can be imported using an absolute import. In the case of `vue-skeleton`, I think the `src` folder is too flexible in that regard, because any developer is allowed to create an arbitrary amount of sub-directories.

There is a middle ground solution, though I'm not exactly sure how I feel about it myself. Instead of using the entire `src` folder in `resolve.modules`, we could `alias` only fixed number of subdirectories (say for example `page`, `util` and `component`)

**Comment by flut1** [comment]
@ReneDrie  @hjeti 

Because this issue hasn't seen any comments for a while, I went ahead and created a pull request to remove absolute imports. If you still want to keep the absolute imports, please respond to the latest arguments in this issue.

During the creation of this commit, I already encountered ambiguous imports. For example, there were absolute imports to both `config` and `modernizr`, which are both widely used [npm](https://www.npmjs.com/package/modernizr) [modules](https://www.npmjs.com/package/config)
One other thing is that we had mixed ordered imports to npm modules and project code (see for example `setupInjects.js`). The common convention is to have the npm imports at the top, and the project imports afterwards. But because we allow for absolute imports to project src, it went unnoticed to both us developers and eslint.

**Comment by hjeti** [comment]
@flut1 It's good that you created a pull request. I am always open for changes in the skeleton if it improves the skeleton or multiple people think it's better this way. 

I didn't know there was a specific order I always use autoimport and let Webstorm decide where the import is added. But keep in mind that import order is important especially in the bootstrap,js. See my code comments on you pr. There are a few issues now because imports from node modules and vue components are placed before other imports.

If you fix those issues I will merge it.

**Comment by flut1** [comment]
@ThaNarie I'm unsure if webpack is even guaranteed to respect your import order (see my comments in the PR). Do you happen to know this? I'm having difficulty to find a conclusive answer online.

**Comment by hjeti** [comment]
@flut1 Why do you need to know that? I fixed multiple issues by changing the import order. Ask @ReneDrie about the css import. At first the screen.css was imported after the other imports but that was causing issues because order is very important in css.

**Comment by ThaNarie** [comment]
Yeah those CSS bundling order issues are super nasty. As far as I could see webpack keeps the import order the same, and executes everything sync.

It would be better if we could find another solution, but for now we should restore the import order for important files, and document the reason in a comment.

**Comment by flut1** [comment]
@hjeti The import order thing probably belongs in a separate issue/PR. I'll update the import order for now to match your comments so we can wrap this PR up

**Comment by hjeti** [comment]
@flut1 Merged your PR and created a new minor version
